### ISSUE TITLE
bug: `single-export-module` is missing types for `defineCustomElement` functions

### ISSUE THREAD

**Issue by weaintplastic** [ionic-team/stencil#4093]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/stencil/blob/main/.github/CONTRIBUTING.md).
- [X] I agree to follow the [Code of Conduct](https://github.com/ionic-team/stencil/blob/main/CODE_OF_CONDUCT.md).
- [X] I have searched for [existing issues](https://github.com/ionic-team/stencil/issues) that already report this problem, without success.

### Stencil Version

3.0.1

### Current Behavior

When using the `dist-custom-element` output target with `customElementsExportBehavior: 'single-export-module'` we are experiencing missing type declarations for the `defineCustomElement...` functions of our components.

```
export { MyComponent as MyComponent } from '../types/components/my-component/my-component';
```

### Expected Behavior

When publishing as a component library, I'd expect that consumers can access the type for the `defineCustomElement...` function the same way they can access the component's type.

```
export { MyComponent as MyComponent, defineCustomElement as defineCustomElementMyComponent  } from '../types/components/my-component/my-component';
```

### Steps to Reproduce

I've created a stackblitz reproduction example here: https://stackblitz.com/edit/node-mhjzxw?file=stencil.config.ts,dist%2Fcomponents%2Findex.js,dist%2Fcomponents%2Findex.d.ts

### Code Reproduction URL

https://stackblitz.com/edit/node-mhjzxw?file=stencil.config.ts,dist%2Fcomponents%2Findex.js,dist%2Fcomponents%2Findex.d.ts

**Comment by tanner-reits** [comment]
Hey @weaintplastic 👋 

Took a look at this and can confirm there appears to be an issue with how/where the component type declarations are getting re-exported. I'll get this labeled and ingested into our backlog for us to resolve. Thanks for reporting!

**Comment by weaintplastic** [comment]
Thanks a lot @tanner-reits

**Comment by tanner-reits** [comment]
@weaintplastic Just wanted to let you know that a fix for this was merged in #4194 and will be available in an upcoming Stencil release! So, keep an eye out for that in the near future. Thanks again for letting us know about this!

**Comment by rwaskiewicz** [comment]
The fix for this issue has gone out as a part of the [Stencil v3.2.1 release](https://github.com/ionic-team/stencil/releases/tag/v3.2.1) - should it reappear, please feel free to open a new issue. Thanks!
